src/components/MessageList/utils/getGroupStyles.ts

```diff
@@ -39,42 +45,46 @@ export const getGroupStyles = <
 >(
   params: GetGroupStylesParams<At, Ch, Co, Ev, Me, Re, Us>,
 ) => {
-  const { messages, noGroupByUser } = params;
+  const {
+    dateSeparators,
+    hideDateSeparators,
+    maxTimeBetweenGroupedMessages,
+    messages,
+    noGroupByUser,
+    userId,
+  } = params;
   const messageGroupStyles: { [key: string]: GroupType[] } = {};
 
-  for (let i = 0; i < messages.length; i++) {
-    const previousMessage = messages[i - 1];
-    const message = messages[i];
-    const nextMessage = messages[i + 1];
-    const groupStyles: GroupType[] = [];
+  const messagesFilteredForNonUser = messages.filter(
+    (message) => !message.deleted_at || userId === message.user?.id,
```

Comment:
`userId` here comes from `client.userID` which is possibly undefined so the 2nd part of this could be true with `undefined === undefined`. would this case result in the filter we want?

Reply:
Yeah thought the same when writing it, but it should have little to no effect on the edge cases like that. Would be no different than it is now.